mozdownload/scraper.py

```diff
@@ -892,7 +891,13 @@ def get_build_info_for_index(self, build_index=None):
 
         # If no index has been given, set it to the last build of the day.
         if build_index is None:
-            build_index = len(parser.entries) - 1
+            # Find the most recent non-empty entry.
+            build_index = len(parser.entries)
+            for build in reversed(parser.entries):
```

Comment:
Could you maybe explain, why you are doing it this particular way? I only program as a hobby, so this may be an approach that is very common, but I have not seen it before.

In your `for` loop, you create a list that is a reversed `parser.entries`, and then you take `build_index` and count down from there.

Why don't you use `enumerate` and get the index this way and then only check for `is_build_dir`?

Sorry for asking such a silly question, but it took me a bit to figure out what you were doing and I was wondering why. Maybe I am also missing something and I can learn something here...